mock_timer_test.go

```diff
@@ -0,0 +1,334 @@
+package glock
```

Comment:
This uses sub tests instead of multiple top-level tests. It's not quite what the rest of the project is using but I think these are easier to read and I imagined the rest of the tests would use them if they existed when we originally wrote them. :D